### PR TITLE
Update docstring for `isnan`

### PR DESCRIPTION
**Pull request by mschauer**
This is defined not only for floating point values, but for all `Number`s in base. Following roughly the Mathworld definition.

### CODE REVIEW

File: base/float.jl [hunk 1]
```diff
@@ -530,7 +530,8 @@ abs(x::Float64) = abs_float(x)
 """
     isnan(f) -> Bool
 
-Test whether a floating point number is not a number (NaN).
+Test whether a number value is a NaN, an indetermined value which is neither an infinity
+nor a finite number ("not a number").
```
Comment (mschauer) [inline base/float.jl]:
```suggestion
nor a finite number ("not a number").
```

File: base/float.jl [hunk 2]
```diff
@@ -530,7 +530,8 @@ abs(x::Float64) = abs_float(x)
 """
     isnan(f) -> Bool
 
-Test whether a floating point number is not a number (NaN).
+Test whether a number value is a NaN, an indetermined value which is neither an infinity
```
Comment (ararslan) [inline base/float.jl]:
```suggestion
Test whether a number value is a NaN, an indeterminate value which is neither an infinity
```

File: base/float.jl [hunk 3]
```diff
@@ -530,7 +530,8 @@ abs(x::Float64) = abs_float(x)
 """
     isnan(f) -> Bool
 
-Test whether a floating point number is not a number (NaN).
+Test whether a number value is a NaN, an indeterminate value which is neither an infinity
```
Comment (nickrobinson251) [inline base/float.jl]:
```suggestion
Test whether a number value is a `NaN`, an indeterminate value which is neither an infinity
```

Comment (mschauer) [reply base/float.jl]: I did not do this because ˋNaNˋ would be Julia code and then explicitly the ˋFloat64ˋ value ˋNaNˋ

Comment (nickrobinson251) [reply base/float.jl]: Ah, didn't realise NaN was Float64. :+1:

**Pull request merged**